src/io/io_entry.c

```diff
@@ -363,26 +364,34 @@ int io_entry_write_fields(const char *state, const char *path,
         fprintf(stream, "\"sha512\":\"%s\",", buf);
     }
 
+    /*
+     * up till this point we have been appending commas to the end of the
+     * entries, this is because we don't know which hashes are going to be
+     * provided. Now that pathmd5 has been put on the stream we switch to
+     * commas before any new entries, allowing errors to simply exit and not
+     * output malformed json.
+     */
+
     unpack(buf, pathmd5, MD5_DIGEST_LENGTH);
-    fprintf(stream, "\"pathmd5\":\"%s\",", buf);
+    fprintf(stream, "\"pathmd5\":\"%s\"", buf); /* no commas before or after */
 
     if (st != NULL)
     {
         /* mode, size and timestamps */
         fprintf(stream,
-                "\"uid\":%u,\"gid\":%u,"
+                ",\"uid\":%u,\"gid\":%u,"
                 "\"mode\":%u,\"size\":%jd,"
                 "\"asec\":%ld,\"ansec\":%ld,"
                 "\"msec\":%ld,\"mnsec\":%ld,"
-                "\"csec\":%ld,\"cnsec\":%ld,",
+                "\"csec\":%ld,\"cnsec\":%ld",
                 st->st_uid, st->st_gid,
```

Comment:
Move to another patch. In fact, why are there some commas at the beginning and some at the end and one line has both. Let's just change all the lines to include the commas at the beginning.